### DISCUSSION TITLE
backup_sram: Needs backup registers SRAM examples for nucleo_U575zi_q

### DISCUSSION THREAD

**Discussion by Niranjan-eaton** [zephyrproject-rtos/zephyr#59223] [Q&A]
**Is your enhancement proposal related to a problem? Please describe.**
backup_sram: Needs backup registers SRAM examples for nucleo_U575zi_q

**Describe the solution you'd like**
Want to run example for  backup SRAM registers,  for nucleo_U575zi_q

**Additional context**

**Comment by erwango** [comment]
Please have a look to `samples/boards/stm32/backup_sram`.
Converting to discussion

**Reply by Niranjan-eaton** [reply]
Yes, looked into samples/boards/stm32/backup_sram, this is for nucleo_h743zi.

looking backup_sram sample example for nucleo_U575zi_q

**Reply by Niranjan-eaton** [reply]
We have sample backup SRAM Example for U575 controller, attached screen of backup SRAM example,
I tried same code, able to build and flash, tested not working as expected.
![image](https://github.com/zephyrproject-rtos/zephyr/assets/114387272/d1ccea9f-a5b3-4e85-8cd0-35286478673c)

but FRASTM, tested and updated the expected results, but I'm not getting correct results, attached screen below
![image](https://github.com/zephyrproject-rtos/zephyr/assets/114387272/69e4bee8-9e86-454c-93ed-c9ae88203df8)

@erwango Please suggest/help on this backup SRAM support for U575 controller

**Comment by Niranjan-eaton** [comment]
U5 controller: Nucleo_U575zi_q
Taking 
In build, memory regions of SRAM's are not showing, 
![image](https://github.com/zephyrproject-rtos/zephyr/assets/114387272/d72db7eb-7d0d-4f3b-98a2-b09a375be8c7)

H7 controller: It was working, in build backup SRAM1,2,3,4 registers are showing 
![image](https://github.com/zephyrproject-rtos/zephyr/assets/114387272/8720c62e-0443-4080-a0e1-9fd5ab7f0ccb)

Give me suggestion, memory regions of SRAM are not showing in U5 controller.

Taken below changes for U5
![image](https://github.com/zephyrproject-rtos/zephyr/assets/114387272/bcd85993-0be2-45b9-8567-47c5f4000d5f)

![image](https://github.com/zephyrproject-rtos/zephyr/assets/114387272/3099e676-cbee-4e1e-9e8b-9d6e0cad0ce0)
![image](https://github.com/zephyrproject-rtos/zephyr/assets/114387272/7e10e8c8-0125-4bc6-9355-d96af1d94f0e)
![image](https://github.com/zephyrproject-rtos/zephyr/assets/114387272/1f1d3065-ba33-47c7-a0e1-e24bbb41de6a)

**Comment by Niranjan-eaton** [comment]
Hi @erwango 
I have tested Zephyr 3.4 setup for U575 controller, test results are not working as expected.
![image](https://github.com/zephyrproject-rtos/zephyr/assets/114387272/d3eeed40-6f47-4596-86cd-c542ce74a01d)

**Reply by erwango** [reply]
@Niranjan-eaton I've just tested and this is working as expected, with no modification on my side on V3.4, using `nucleo_u575zi_q`:
```
*** Booting Zephyr OS build zephyr-v3.4.0-416-ga44f61c5f54d ***
Current value in backup SRAM (0x40036400): 1
Next reported value should be: 2
Keep VBAT power source and reset the board now!
*** Booting Zephyr OS build zephyr-v3.4.0-416-ga44f61c5f54d ***
Current value in backup SRAM (0x40036400): 2
Next reported value should be: 3
Keep VBAT power source and reset the board now!

```

**Reply by Niranjan-eaton** [reply]
Hi @erwango 
Thanks for reply.

While  testing  SRAM backup register sample, need  any H/W configuration like related **VBAT pin** configurations for U5 ?
Any **dependency on VBAT**, while testing SRAM backup register sample ?

**Reply by erwango** [reply]
Hi @Niranjan-eaton, I just see that I made an error in the comment above. I actually used b_u585i_iot01a, not the nucleo board, sorry for the confusion. Maybe the difference of status lies in there
